hera_mc/geo_handling.py

```diff
@@ -313,33 +322,292 @@ def get_location(self, to_find, query_date, show_location=False, verbosity='m'):
+        for k, stn_type in self.station_types.iteritems():
+            if station_types_to_check == 'all' or k in station_types_to_check:
+                for stn in stn_type['Stations']:
+                    print("Checking {}".format(stn))
+                    # Get First one
+                    ustn = stn.upper()
+                    conn = self.session.query(part_connect.Connections).filter(
+                        func.upper(part_connect.Connections.upstream_part) == ustn).order_by(
+                        asc(part_connect.Connections.start_gpstime)).first()
+                    if conn is None:
+                        continue
                     conn.gps2Time()
-                    stations_conn.append({'station_name': stn.station_name,
-                                          'station_type': stn.station_type_name,
-                                          'longitude': stn.lon,
-                                          'latitude': stn.lat,
-                                          'elevation': stn.elevation,
-                                          'antenna_number': ant_num,
-                                          'start_date': conn.start_date,
-                                          'stop_date': conn.stop_date})
-        return stations_conn
+                    if conn.start_date < earliest_date:
+                        at_date = earliest_date
+                    else:
+                        at_date = conn.start_date + TimeDelta(3600.0, format='sec')
+                    fc = cm_revisions.get_full_revision(stn, at_date, full_req, self.session)
+                    if len(fc) == 0 and at_date < now:
+                        fc, at_date = self._search_loop(stn, at_date, now, m, base_tdelta, full_req)
+                    if len(fc) == 0:
+                        continue
+                    station_dict = self.get_fully_connected_location_at_date(stn=stn, at_date=at_date,
+                                                                             hookup=hookup, fc=fc, full_req=full_req)
+                    station_conn.append(station_dict)
+                    # Get subsequent ones
+                    k0 = fc[0].hookup['timing'].keys()[0]
+                    tp = fc[0].hookup['timing'][k0]
+                    fc_strt = Time(cm_utils.get_date_from_pair(tp['e'][0], tp['n'][0], 'earliest'), format='gps')
+                    fc_stop = cm_utils.get_date_from_pair(tp['e'][1], tp['n'][1], 'earliest')
+                    if fc_stop is None or fc_stop > now.gps:
+                        continue
+                    fc_stop = Time(fc_stop, format='gps')
+                    at_date = fc_stop + TimeDelta(base_tdelta, format='sec')
+                    while fc_stop is not None:
+                        fc, at_date = self._search_loop(stn, at_date, now, m, base_tdelta, full_req)
+                        if len(fc) == 0:
+                            fc_stop = None
+                        else:
```

Comment:
I believe this can be changed to:

```python
if not len(fc):
    fc_stop = None
    break

station_dict = ...
```

Once again this reduces the amount of rightward drift in the code.